daemon/c_smartcard.c

```diff
@@ -930,11 +930,17 @@ c_smartcard_token_detach(void *smartcardp)
 	c_smartcard_t *smartcard = smartcardp;
 	ASSERT(smartcard);
 
-	DEBUG("USB token has been detached, stopping Container %s",
-	      container_get_name(smartcard->container));
+	// check if the container is already stopped/stopping
+	if (container_is_stoppable(smartcard->container)) {
```

Comment:
check in compartment_stop is sufficient -> remove here

Reply:
currently the caller of container_stop/start has to check if the container is startable/stopable. Should we move this check to the compartment_start/stop function instead and remove all caller-checks?

Reply:
I could either adapt the code, so that no caller-check is required (as asked for here), or I could adapt it, that every call has to check if the container is startable/stopable (as in cmld_container_stop/start).

I think enforcing the check in the compartment_stop/start results in cleaner code. @quitschbo @k0ch4lo